include/deal.II/base/symmetric_tensor.h

```diff
@@ -915,6 +915,17 @@ namespace internal
 
 template <int rank, int dim, typename Number>
 inline
+SymmetricTensor<rank,dim,Number>::SymmetricTensor ()
+{
+  // Some auto-differentiable numbers need explicit
+  // zero initialization.
+  for (unsigned int i=0; i<base_tensor_type::dimension; ++i)
+    data[i] = internal::NumberType<Number>::value(0.0);
```

Comment:
I missed this the first time around -- for rank-4 tensors, `data` is a rank-2 tensor. Can it be initialized this way? I think you may want to try creating a `SymmetricTensor<4,dim,...some AD type...>` or maybe just using a `std::complex` type. That would probably also make for a good stand-alone testcase.